drivers/wifi/nrfwifi/Kconfig.nrfwifi

```diff
@@ -0,0 +1,667 @@
+# Nordic Wi-Fi driver for nRF70 series SoCs
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# Hidden symbols for internal use
+config NRF7002
+	bool
+	default y if DT_HAS_NORDIC_NRF7002_SPI_ENABLED || DT_HAS_NORDIC_NRF7002_QSPI_ENABLED
+
+config NRF7001
+	bool
+	default y if DT_HAS_NORDIC_NRF7001_SPI_ENABLED || DT_HAS_NORDIC_NRF7001_QSPI_ENABLED
+
+config NRF7000
+	bool
+	default y if DT_HAS_NORDIC_NRF7000_SPI_ENABLED || DT_HAS_NORDIC_NRF7000_QSPI_ENABLED
```

Comment:
These values should be within 
```
if WIFI_NRF70

...

endif
```

Reply:
Hmm, this will bleed into .config whether NRF70 exists or not if WIFI is selected, IMO it is better to directly depend on the configs from the devicetree in `menuconfig WIFI_NRF70` using the following pattern:
```
menuconfig WIFI_NRF70
        default y
        depends on DT_HAS_NORDIC_NRF7002_SPI_ENABLED || \
                   DT_HAS_NORDIC_NRF7002_QSPI_ENABLED || \
                   ...
```
since this keeps them contained, @kartben @fabiobaltieri  WDYT?

Reply:
These are used in multiple places, so, was trying to avoid repetition by using a simpler hidden config.

Reply:
I think it should be as @bjarki-andreasen said, so that you cannot accidentally enable the option if none of the two is available, as it is now because the condition is a `default ... if` rather than `depends on`.

What repetition did you want to avoid? the `SPI || QSPI`? You could have an intermediate symbol for that alone.

Reply:
Does this work?

```
menuconfig WIFI_NRF70
	bool "nRF70 driver"
	select NET_L2_WIFI_MGMT if NETWORKING
	select NET_L2_ETHERNET_MGMT if NETWORKING && NET_L2_ETHERNET
	select WIFI_USE_NATIVE_NETWORKING if NETWORKING
	select EXPERIMENTAL if !SOC_SERIES_NRF53X && !SOC_SERIES_NRF91X
	default y
	depends on \
		DT_HAS_NORDIC_NRF7002_SPI_ENABLED || DT_HAS_NORDIC_NRF7002_QSPI_ENABLED || \
		DT_HAS_NORDIC_NRF7001_SPI_ENABLED || DT_HAS_NORDIC_NRF7001_QSPI_ENABLED || \
		DT_HAS_NORDIC_NRF7000_SPI_ENABLED || DT_HAS_NORDIC_NRF7000_QSPI_ENABLED
	help
	  Nordic Wi-Fi Driver

# Hidden symbols for internal use
config NRF7002
	bool
	default y if DT_HAS_NORDIC_NRF7002_SPI_ENABLED || DT_HAS_NORDIC_NRF7002_QSPI_ENABLED

config NRF7001
	bool
	default y if DT_HAS_NORDIC_NRF7001_SPI_ENABLED || DT_HAS_NORDIC_NRF7001_QSPI_ENABLED

config NRF7000
	bool
	default y if DT_HAS_NORDIC_NRF7000_SPI_ENABLED || DT_HAS_NORDIC_NRF7000_QSPI_ENABLED
```

Reply:
I'd say so

Reply:
With the prefix its perfect :)